Source/ProgressionSystemRuntime/Private/Data/PSSaveGameData.cpp

```diff
@@ -2,16 +2,17 @@
 
 #include "Data/PSSaveGameData.h"
 
+#include "Data/PSDataAsset.h"
 #include "Data/PSWorldSubsystem.h"
 #include "UtilityLibraries/MyBlueprintFunctionLibrary.h"
 
 #include UE_INLINE_GENERATED_CPP_BY_NAME(PSSaveGameData)
 
 // Retrieves the name of the save slot, safely initializing the name statically to ensure thread safety and initialization order.
-const FString& UPSSaveGameData::GetSaveSlotName()
+const FString& UPSSaveGameData::GetSaveSlotName(int32 SaveSlotVersion)
 {
 	// Using a function-static variable to avoid the static initialization order fiasco
-	static const FString SaveSlotName = StaticClass()->GetName();
+	static const FString SaveSlotName = FString::Printf(TEXT("%s-%d"), *StaticClass()->GetName(), SaveSlotVersion);
```

Comment:
1. **Remove `static` from local variable, make it just `const FString`**
SaveSlotName local variable can not be `static` anymore, since from now it's combinted with printf in runtime. Smarter compilier such as clang on linux\ubuntu might notice this mistake and throw error.

2. **Please refactor function to return `FString`-copy (without &), but not `const FString&`**
Function can not return `const ref` anymore, but only copy, for the same reason as above - it's dynamic, so it will not compile. 

Expected result:
```cpp
FString UPSSaveGameData::GetSaveSlotName(int32 SaveSlotVersion)
{
const FString SaveSlotName = ...
// ...
}
```

Reply:
done

Reply:
@h4rdmol, is not done yet.
- Please remove `const` from return, no sense to return const-copy. Just return FString. See my example above
- also, it seems you dont need local variable anymore, just put this logic right after `return FString::Printf...`

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/44/commits/1a99b6e948f93733912803fda80a05750d6d5bf8